### ISSUE TITLE
Tomb Raider 2013 Incorrect rendering

### ISSUE THREAD

**Issue by rox4d** [iXit/Mesa-3D#324]
with gm107 (nouveau with patch from https://lists.freedesktop.org/archives/mesa-dev/2018-November/210359.html) it looks like upscaled from lower resolution with superfluous red lines across a screen.
d3d9 trace from windows:
https://drive.google.com/file/d/1CGd-CR5vQmwe4jy4Fsux-jDypJUHJP7-/view
not sure if it is useful, but replaying with nine show a problem.

**Comment by iiv3** [comment]
It seems like another nouveau problem, as the trace plays perfectly with R600 driver (HD5670 Evergreen Redwood).

Have you tried playing the trace without the samplers patch? Do you get the same problems. Aka are these problems existing or caused by the patch?

**Comment by rox4d** [comment]
yes, It is plays fine with R600 (rv730).

**Comment by orbea** [comment]
Is this fixed with the mesa master now?

**Comment by rox4d** [comment]
No, this report is not about nouveau d3d9 regression (just fixed).
![screenshot](https://user-images.githubusercontent.com/34712490/49342017-36b92300-f666-11e8-9880-678415938a2a.png)

**Comment by rox4d** [comment]
I was not trying it then.
This trace was made with antialiasing disabled in settings.

**Comment by rox4d** [comment]
I was compiling several older version mesa (from git before nine was broken with nouveau) to check whether it regression, but it seems that d3dadapter9 was not good enough there (crash). Binary mix from old mesa dri + (d3dadapter9.so.1.0.0 from newer git + nouveau workaround) looks like screenshot.

**Comment by orbea** [comment]
A nouveau developer was looking at this the other day as discussed in irc, but he also ran into crashes. Maybe it was the same issue?

**Comment by rox4d** [comment]
Who knows:), for me d3dretrace.exe crashes while trying to replay this  trace with nine disabled (wine-staging-nine-3.18).

**Comment by orbea** [comment]
The crashes when replaying the trace were also discussed in irc, I think diagnosing and potentially fixing the crashes when replaying the trace or starting the game would make it easier to fix this issue.

**Comment by rox4d** [comment]
The game is works  for me in wine (windows version steam installed), in winecfg OS is Windows XP, and these dlls are disabled: d3d10, d3d10_1, d3d10core, d3d11, dxgi. The registry key "HKEY_CURRENT_USER\Software\ Crystal Dynamics\Tomb Raider\Graphics\RenderAPI"  are equal 9.
